saleor/graphql/discount/tests/test_sale_channel_listing.py

```diff
@@ -0,0 +1,154 @@
+from unittest.mock import patch
+
+import graphene
+
+from ....discount.models import SaleChannelListing
+from ...tests.utils import assert_negative_positive_decimal_value, get_graphql_content
+
+SALE_CHANNEL_LISTING_UPDATE_MUTATION = """
```

Comment:
Lack of test for :
- duplicated IDs in add 
- duplicated IDs in remove 
- Same id in add and remove
- invalid decimal precision

Reply:
In these tests, we should add custom param to errors to identify witch input course error.